docs/developer/proposal/EP_004-ACPI-support.md

```diff
@@ -0,0 +1,292 @@
+└── exporter/prometheus/collector/
+    └── platform_collector.go  # Platform power metrics collector
+```
+
+### API/Interface Changes
+
+[Describe any changes to public APIs, service interfaces, or data structures. Show code snippets for new or modified interfaces.]
+
+```go
+// acpi_power_meter.go - Implements powerMeter interface
+type acpiPowerMeter struct {
+    logger *slog.Logger
+    devicePath string
+}
+
+// Service for ACPI power meter
+type Service struct {
+    logger *slog.Logger
+    powerMeter acpiPowerMeter
+}
+```
+
+## Configuration
+
+### Main Configuration Changes
+
+`ACPI.Enabled` is False by default.
+
+```go
+type Platform struct {
+    Redfish Redfish `yaml:"redfish"`
+    ACPI    ACPI    `yaml:"acpi"`
+}
+
+type ACPI struct {
+    Enabled    *bool  `yaml:"enabled"`
+}
+```
+
+**CLI flags**
+
+```
+--platform.acpi.enabled=true  # Enable ACPI monitoring
+```
+
+### Configuration File (if applicable)
+
+```yaml
+# add lines to /etc/kepler/config.yaml
+platform:
+  acpi:
+    enabled: false
+```
+
+### Security Considerations
+
+This requires sysfs read privilege which is originally requested by RAPL part.
+
+## Deployment Examples
+
+### Kubernetes Environment
+
+```yaml
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: kepler
+spec:
+  template:
+    spec:
+      containers:
+      - name: kepler
+        args:
+        - --kube.enable=true
+        - --kube.node-name=$(NODE_NAME)
+        - --platform.acpi.enabled=true
+        env:
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+```
+
+### Standalone Deployment
+
+```bash
+# Run Kepler with ACPI support
+./kepler --platform.acpi.enabled=true
+```
+
+## Testing Strategy
+
+### Test Coverage
+
+- **Unit Tests**:
+  - Verify acpiPowerMeter correctly reads and parses a single power1_average file.
+  - Test aggregation of multiple power meters (power1_average, power2_average) into a single metric.
+  - Simulate sysfs file read failures and verify error handling.
+  - Test calculation of `kepler_platform_joules_total` by multiplying wattage with elapsed time.
+- **Integration Tests**:
+  - Validate that enabling `ACPI.Enabled=true` initializes the ACPI power meter service.
+  - Ensure metrics are correctly labeled with `source="acpi"` and exposed via the Prometheus exporter.
+- **E2E Tests**:
+  - Deploy Kepler in a Kubernetes cluster and verify metrics are available in Prometheus.
+  - Test a node without ACPI support and ensure Kepler falls back to other available collectors (e.g., RAPL).
+
+### Test Infrastructure
+
+- Testdata Structure: Mimic real sysfs structure under `internal/device/platform/acpi/testdata/`
+  - Fake multiple power meters (`ACPI000D:xx`) or multiple measurements (`power*_average`) for testing
+  - e.g., `testdata/sys/bus/acpi/drivers/power_meter/ACPI000D:00/power1_average`, with sample values for power readings (e.g., 450500000 microwatts for 450.5 watts).
```

Comment:
Can there be a different unit as well? Eg. milliwatts instead of micro?  We have had case where the power reported by kepler was abnormally high .. 
See: https://github.com/sustainable-computing-io/kepler/issues/1774 -> https://github.com/sustainable-computing-io/kepler-metal-ci/blob/main/docs/train-validate-e2e/2024-09-09_22-41-38/AbsPower/BPFOnly/ExponentialRegressionTrainer/report-v0.7.11-212-gac5ee8b8.md#platform---idle

Reply:
https://github.com/torvalds/linux/blob/master/Documentation/hwmon/sysfs-interface.rst#power does say `microWatt`